provd/proto/identity.proto

```diff
@@ -0,0 +1,35 @@
+syntax = "proto3";
+
+package identity;
+
+service IdentityService {
+  rpc GetIdentity(Empty) returns (GetIdentityResponse);
+  rpc SetIdentity(SetIdentityRequest) returns (Empty);
+  rpc ValidateUsername(ValidateUsernameRequest) returns (ValidateUsernameResponse);
+}
+
+message Empty {}
+
+message Identity {
+  string realName = 1;
+  string hostname = 2;
+  string username = 3;
+  string password = 4;
+  bool autoLogin = 5;
+}
+
+message GetIdentityResponse {
+  Identity identity = 1;
+}
+
+message SetIdentityRequest {
+  Identity identity = 1;
+}
+
+message ValidateUsernameRequest {
+  string username = 1;
+}
+
+message ValidateUsernameResponse {
+  bool validation = 1;
```

Comment:
'valdiation' could be ambiguous as a boolean. For example, does `validation: false` mean validation didn't run or that the input wasn't valid?

Maybe simply: `bool valid = 1;`?